pkg/registry/generic/registry/store.go

```diff
@@ -227,6 +230,22 @@ func (e *Store) Create(ctx api.Context, obj runtime.Object) (runtime.Object, err
 	if err := e.Storage.Create(ctx, key, obj, out, ttl); err != nil {
 		err = storeerr.InterpretCreateError(err, e.QualifiedResource, name)
 		err = rest.CheckGeneratedNameError(e.CreateStrategy, err, obj)
+		if !storage.IsNodeExist(err) && e.OnCreateFailure == nil {
+			return nil, err
+		}
+		if errGet := e.Storage.Get(ctx, key, out, false); errGet != nil {
+			return nil, storeerr.InterpretGetError(errGet, e.QualifiedResource, name)
+		}
+		accessor, errGetAcc := meta.Accessor(out)
+		if errGetAcc != nil {
+			return nil, kubeerr.NewInternalError(err)
+		}
+		if accessor.GetDeletionTimestamp() != nil {
+			err.(*kubeerr.StatusError).ErrStatus.Message += "- the object may be in the process of deletion!\n"
```

Comment:
Message wise, don't add newlines or exclamation points.  You also don't need to qualify the error  with `may`, because it is:

```
message = fmt.Sprintf("pod is being deleted: %s", message)
```

would be more succinct.